### PR TITLE
Standardise SVG element structure, upgrade gridlines + misc

### LINKED ISSUE #255: Standardise the SVG element structure
We should either -
* Always have a root element appended to the selection by the component to which all other in the component. This root element will have a CSS class and all CSS will use a descendant selector on that class name.
* Never add a root element. All CSS should target CSS classes as appropriate.

### PR DESCRIPTION
**Pull request by chrisprice**
I also deleted the README.md files as they're out of date.

Fixes #255, #257, #232.

### CODE REVIEW

**Comment by chrisprice** [comment]
Also #139

File: components/fc.css [hunk 1]
```diff
@@ -18,3 +18,13 @@
 .axis text { 
   font: 10px sans-serif; 
 }
+
+.gridlines { 
+  stroke: #ccc;
+  stroke-width: 1.0;
+}
+
+.chartArea rect.background {
```
Comment (ColinEberhardt) [inline components/fc.css]: I guess whether we keep this or not depends on the future of chartLayout? Still worth merging this PR all the same

File: components/scale/linear.js [hunk 2]
```diff
@@ -1,72 +0,0 @@
-(function(d3, fc) {
```
Comment (ColinEberhardt) [inline components/scale/linear.js]: cool - I was intending to delete this scale ...

**Comment by ColinEberhardt** [comment]
Thanks @chrisprice - it's a :+1: from me.

**Pull request merged**